lib/resolveurl/plugins/mixdrop.py

```diff
@@ -0,0 +1,48 @@
+"""
+    OVERALL CREDIT TO:
+        t0mm0, Eldorado, VOINAGE, BSTRDMKR, tknorris, smokdpi, TheHighway
+
+    urlresolver XBMC Addon
+    Copyright (C) 2011 t0mm0
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import re
+from urlresolver import common
+from urlresolver.resolver import UrlResolver, ResolverError
+from lib import jsunpack
+
+class MixdropResolver(UrlResolver):
+    name = "mixdrop"
+    domains = ["mixdrop.co"]
+    pattern = '//(mixdrop\.co)/(?:f|e)/(.+)'
+
+    def __init__(self):
+        self.net = common.Net()
+
+    def get_media_url(self, host, media_id):
+        web_url = self.get_url(host, media_id)
+        headers = {'User-Agent': common.RAND_UA}
+        html = self.net.http_GET(web_url, headers=headers).content
+
+        try:
+            r = re.search('\s+?(eval\(function\(p,a,c,k,e,d\).+)\s+?', html)
+            r = jsunpack.unpack(r.group(1))
+            r = re.search("vsrc=\"([^\"]+)", r.replace('\\', ''))
+            return "https:" + r.group(1)
+        except:
```

Comment:
Please avoid using try-except clauses where possible as it makes hard when debugging later
See https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py for example on how to write the same look for  (p,a,c,k,e,d) block, unpack and look for vsrc using if clause

Reply:
How exactly is that easier to debug? It'll yield the same ResolverError exception and no additonal output helpful to any debugging means.

In both cases one would need to add some output in order to debug or am I missing something?

Reply:
if it is a try-except clause, there is no failure, so nothing in the logs
however with an if condition, when it fails, the log will have the line number where it failed, so easy to debug

Reply:
Can you elaborate on which kind of failure would log a message in your example? Aren't the potential sources of failure caught in an if clause there?

Reply:
with an 'if' clause, in case line 42 fails, there will be an error in the log for line 43
however with try/except, when line 42 fails it will switch to except clause and continue, so no error in log file

Reply:
In your example, if https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py#L40 fails, it'll jump straight to https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py#L46 and you still don't know if https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py#L40 or https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py#L43 failed because you'll just get an exception for the exception raised at https://github.com/jsergio123/script.module.resolveurl/blob/master/lib/resolveurl/plugins/onlystream.py#L46

Reply:
And you'll get the exact same exception at https://github.com/jsergio123/script.module.resolveurl/blob/78c405cfb4a0364e9154e2e171d89034fba54776/lib/resolveurl/plugins/mixdrop.py#L39

So there's really no difference? In both cases, yours and mine, you just know something failed but not what exactly.

Reply:
I am not saying that my example is better, all I said was if possible avoid using try/except clause in the resolver plugins. It is just a matter of coding style, if you want to use try/except then let it be

Reply:
I just tried to understand what you meant with yours is easier to debug because I couldn't see the difference you spoke about. But now that you agree there is none, we can leave it like this.